pkg/chain/ethereum/config.go

```diff
@@ -39,12 +39,12 @@ type Config struct {
 	// time, the gas price is increased and transaction is resubmitted.
 	MiningCheckInterval int
 
-	// MaxGasPrice specifies the maximum gas price the client is
+	// MaxGasPrice specifies the maximum gas price (in wei) the client is
 	// willing to pay for the transaction to be mined. The offered transaction
 	// gas price can not be higher than the max gas price value. If the maximum
 	// allowed gas price is reached, no further resubmission attempts are
 	// performed.
-	MaxGasPrice uint64
+	MaxGasPrice *Weis
```

Comment:
One last comment: I think it should be `Wei`. "The gas price is 20 Gwei", "The gas price is 20000000000 wei" and not "... 20 Gweis` or "... 20000000000 weis".

Reply:
`Weis` was a workaround for a collision with a `Wei` unit. As we made the unit private we can now rename the struct to `Wei`.